### ISSUE TITLE
NXOS-SSH get_mac_address_table() error

### ISSUE THREAD

**Issue by Rian83** [napalm-automation/napalm#524]
Hello Guys,

When I try to get the mac add table from a Nexus 5020 (version 5.2(1)N1(9b)).
Do I miss something ?

```
from napalm import get_network_driver
driver = get_network_driver('nxos_ssh')


device=driver('AAAA', 'censured', 'censured')
device.open()
out=device.get_mac_address_table()
```

> Traceback (most recent call last):
>   File "script.py", line 8, in <module>
>     out=device.get_mac_address_table()
>   File "/usr/local/lib/python2.7/dist-packages/napalm/nxos_ssh/nxos_ssh.py", line 1318, in get_mac_address_table
>     raise ValueError("Unexpected output from: {}".format(repr(line)))
> ValueError: Unexpected output from: u'+ 96       000e.1e49.bc22    dynamic   0          F    F  Po1'
>

**Comment by bewing** [comment]
Can you paste us a sanitized output of `show mac address-table` from the device in question?

**Comment by Rian83** [comment]
> HOSTNAME_CENSURED# sh mac address-table
> Legend:
>         * - primary entry, G - Gateway MAC, (R) - Routed MAC, O - Overlay MAC
>         age - seconds since last seen,+ - primary entry using vPC Peer-Link
>    VLAN     MAC Address      Type      age     Secure NTFY   Ports/SWID.SSID.LID
> ---------+-----------------+--------+---------+------+----+------------------
> * 2        0000.0c9f.f002    dynamic   14400      F    F  Po2
> * 2        0026.51c7.45c1    dynamic   110        F    F  Po2
> * 2        0026.9802.8cc1    dynamic   0          F    F  Po2
> * 2        0026.9802.ad41    dynamic   0          F    F  Po2
> * 2        0026.9802.b941    dynamic   60         F    F  Po2
> * 2        0050.5692.05da    dynamic   13920      F    F  Po2
> * 2        0050.5692.0ef6    dynamic   5170       F    F  Po2
> * 2        0050.5692.1056    dynamic   0          F    F  Po2
> * 2        0050.5692.214f    dynamic   3240       F    F  Po2
> * 2        0050.5692.2318    dynamic   2520       F    F  Po2
> * 2        0050.5692.37b9    dynamic   730        F    F  Po2
> * 2        0050.5692.40e9    dynamic   10         F    F  Po2
> * 2        0050.5692.55f9    dynamic   200        F    F  Po2
> * 2        0050.5692.5706    dynamic   130        F    F  Po2
> * 2        0050.5692.5d20    dynamic   700        F    F  Po2
> * 2        0050.5692.5df2    dynamic   4650       F    F  Po2
> * 2        0050.5692.5ea7    dynamic   490        F    F  Po2
> * 2        0050.5692.6402    dynamic   50         F    F  Po2
> * 2        0050.5692.6629    dynamic   980        F    F  Po2
> * 2        0050.5692.6d1a    dynamic   460        F    F  Po2
> * 2        0050.5692.7f1c    dynamic   230        F    F  Po2
> * 2        7281.608f.910f    dynamic   20         F    F  Po2
> * 2        7281.6521.d70f    dynamic   80         F    F  Po2
> * 2        7281.6615.f20f    dynamic   100        F    F  Po2
> * 2        7281.68c2.ba0f    dynamic   100        F    F  Po2
> * 2        7281.6992.b40f    dynamic   20         F    F  Po2
> * 2        98be.9466.f41a    dynamic   20         F    F  Po2
> * 2        98be.9466.f49a    dynamic   0          F    F  Po2
> * 2        98be.9466.f54a    dynamic   20         F    F  Po2
> * 2        98be.9466.f5fe    dynamic   0          F    F  Po2
> * 2        b296.7650.2512    dynamic   20         F    F  Po2
> * 2        e6d2.6373.0f0c    dynamic   100        F    F  Po2
> * 2        ee72.b03f.150f    dynamic   20         F    F  Po19
> * 2        ee72.bd72.170f    dynamic   80         F    F  Po19
> * 2        ee72.bdb5.2e0f    dynamic   80         F    F  Po19
> * 1602     000c.29da.2285    dynamic   20         F    F  Po2
> [...]

And another info for you, if run the script, I get this error : 

> Traceback (most recent call last):
>   File "script.py", line 8, in <module>
>     out=device.get_mac_address_table()
>   File "/usr/local/lib/python2.7/dist-packages/napalm/nxos_ssh/nxos_ssh.py", line 1318, in get_mac_address_table
>     raise ValueError("Unexpected output from: {}".format(repr(line)))
> ValueError: Unexpected output from: u'+ 96       000e.1e49.bc22    dynamic   0          F    F  Po1'

So If I run this command : `sh mac address-table | egrep -B 4 -A 5 000e.1e49.bc22`
Here's a summury of what I get, to show you the context of the "parsing" bug : 

> * 96       000a.f76d.fce6    dynamic   10         F    F  Po2
> * 96       000a.f76e.0706    dynamic   50         F    F  Po2
> * 96       000e.1e48.f080    dynamic   10         F    F  Po2
> * 96       000e.1e49.bc20    dynamic   10         F    F  Eth1/11
> + 96       000e.1e49.bc22    dynamic   0          F    F  Po1
> * 96       000e.1e49.e5b0    dynamic   30         F    F  Po2
> * 96       000e.1e49.e5b2    dynamic   10         F    F  Po2
> * 96       000e.1e4a.1270    dynamic   30         F    F  Po2
> * 96       000e.1e4a.1450    dynamic   10         F    F  Po2
> * 96       000e.1e4a.1452    dynamic   80         F    F  Po2

**Comment by Rian83** [comment]
Maybe the issue came from the '+' at the beginning of the ligne.
From the source code, there is no "test" regarding the '+'

**Comment by Rian83** [comment]
I've fixed the issue by editing this line : 
napalm/nxos_ssh/nxos_ssh.py:1292 : 
`        output = re.sub(r"^[\*G\]", "", output, flags=re.M)`
by
`        output = re.sub(r"^[\*G\+]", "", output, flags=re.M)`

But I still have an issue because I have also the beginning of a line that start with a space : 
```
* 3        f60c.b0d7.f907    dynamic   350        F    F  Po2
* 3        f60c.b9ac.d603    dynamic   0          F    F  Po2
* 3        f60c.bdb8.b607    dynamic   0          F    F  Po2
* 3        f60c.be90.7a07    dynamic   0          F    F  Po2
  20       0100.5e7f.fffd    igmp      0          F    F  Po1 Po2 Po17
  10       0100.5e7f.fffd    igmp      0          F    F  Po1 Po2 Po17
```

**Comment by Rian83** [comment]
For the second issue, I partially resolved the issue by changing this code : 

```
            for pattern in [RE_MACTABLE_FORMAT1, RE_MACTABLE_FORMAT2]:
                if re.search(pattern, line):
                    if len(line.split()) == 7:
                        vlan, mac, mac_type, _, _, _, interface = line.split()
                        mac_address_table.append(process_mac_fields(vlan, mac, mac_type, interface))
                        break
                    elif len(line.split()) > 7:
                        splits = line.split()
                        mac_address_table.append(process_mac_fields(splits[0], splits[1], splits[2], splits[6]))
                        break
```
But the problem is the multiple interfaces. The interface part of the returned list should not be a string but a list.
The bad output : 
`{u'vlan': 20, u'last_move': -1.0, u'active': False, u'mac': u'01:00:5E:7F:FF:FD', u'static': False, u'interface': u'Po1', u'moves': -1}`
The new output : 
`{u'vlan': 20, u'last_move': -1.0, u'active': False, u'mac': u'01:00:5E:7F:FF:FD', u'static': False, u'interface': ['Po1', 'Po2', 'Po17'], u'moves': -1}`

**Comment by ktbyers** [comment]
One part of this is fixed here: https://github.com/napalm-automation/napalm/pull/548

Second part with multiple ports still needs fixed. Do you want to work on a PR on that?

**Comment by Rian83** [comment]
Yes #548 should solved the first issue.
I can't PR because I didn't have git and internet access on my dev server. I only work with the zip file ...

I discover another issue : 
```
* 3        f60c.be90.7a07    dynamic   300        F    F  Po2
  39       0100.5e00.4b4b    igmp      0          F    F  Po1 Po2 Po22
  516      0100.5e00.4b4b    igmp      0          F    F  Po1 Po2 Po14
  110      0100.5e00.0118    igmp      0          F    F  Po1 Po2 Po14
                                                          Po18 Po24
  110      0100.5e01.0203    igmp      0          F    F  Po1 Po2 Po24
  110      0100.5e01.0207    igmp      0          F    F  Po1 Po2 Po24
```
When the list of interfaces are on multiple lines, the line with lot of whitespace will crash.

**Comment by ktbyers** [comment]
Yuck...that is unpleasant.

**Comment by Rian83** [comment]
@ktbyers I've installed git ;-)
I can push my modification if you need

**Comment by Rian83** [comment]
@ktbyers For retrocompatibility, how can we return the list of ports and not only the first one ?
Maybe a string like `{u'vlan': 20, u'last_move': -1.0, u'active': False, u'mac': u'01:00:5E:7F:FF:FD', u'static': False, u'interface': u'Po1,Po2,Po24,Eth1/2', u'moves': -1}`
Otherwise, if we return a list (`{u'vlan': 20, u'last_move': -1.0, u'active': False, u'mac': u'01:00:5E:7F:FF:FD', u'static': False, u'interface': ["Po1", "Po2", "Po24", "Eth1/2"], u'moves': -1}`), we'll brake the compatibility

**Comment by dbarrosop** [comment]
Closed via #524
